File: app/src/main/java/org/wikipedia/views/SearchActionProvider.java
package org.wikipedia.views;

import android.content.Context;
import android.graphics.Color;
import android.view.View;
import android.view.inputmethod.EditorInfo;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.SearchView;
import androidx.core.view.ActionProvider;

import org.wikipedia.R;
import org.wikipedia.databinding.GroupSearchBinding;
import org.wikipedia.util.DeviceUtil;

import static org.wikipedia.util.ResourceUtil.getThemedColor;

public class SearchActionProvider extends ActionProvider {

    public interface Callback {
        void onQueryTextChange(String s);
        void onQueryTextFocusChange();
    }

    private CabSearchView searchView;

    private Context context;
    private String searchHintString;
    private Callback callback;

    public SearchActionProvider(@NonNull Context context, @NonNull String searchHintString, @NonNull Callback callback) {
        super(context);
        this.context = context;
        this.searchHintString = searchHintString;
        this.callback = callback;
    }

    @Override
    public boolean overridesItemVisibility() {
        return true;
    }

    @Override
    public View onCreateActionView() {
        final GroupSearchBinding binding = GroupSearchBinding.bind(View.inflate(context, R.layout.group_search, null));
        searchView = binding.searchInput;
        searchView.setFocusable(true);
        searchView.setIconified(false);
        searchView.setMaxWidth(Integer.MAX_VALUE);
        searchView.setInputType(EditorInfo.TYPE_CLASS_TEXT);
        searchView.setSubmitButtonEnabled(false);
        searchView.setQueryHint(searchHintString);
        searchView.setSearchHintTextColor(getThemedColor(getContext(), R.attr.material_theme_de_emphasised_color));
        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String s) {
                return false;
            }

            @Override
            public boolean onQueryTextChange(String s) {
                searchView.setCloseButtonVisibility(s);
                callback.onQueryTextChange(s);
                return true;
            }
        });

        searchView.setOnQueryTextFocusChangeListener((v, isFocus) -> {
            if (!isFocus) {
                callback.onQueryTextFocusChange();
            }
        });

        // remove focus line from search plate
        View searchEditPlate = searchView.findViewById(androidx.appcompat.R.id.search_plate);
        searchEditPlate.setBackgroundColor(Color.TRANSPARENT);

        DeviceUtil.showSoftKeyboard(searchView);
        return binding.getRoot();
    }
}

File: app/src/main/java/org/wikipedia/views/SearchActionProvider.kt
package org.wikipedia.views

import android.content.Context
import android.graphics.Color
import android.view.LayoutInflater
import android.view.View
import android.view.inputmethod.EditorInfo
import androidx.appcompat.widget.SearchView
import androidx.core.view.ActionProvider
import org.wikipedia.R
import org.wikipedia.databinding.GroupSearchBinding
import org.wikipedia.util.DeviceUtil.showSoftKeyboard
import org.wikipedia.util.ResourceUtil.getThemedColor

class SearchActionProvider(context: Context,
                           private val searchHintString: String,
                           private val callback: Callback) : ActionProvider(context) {
    interface Callback {
        fun onQueryTextChange(s: String)
        fun onQueryTextFocusChange()
    }

    private val binding = GroupSearchBinding.inflate(LayoutInflater.from(context))

    override fun onCreateActionView(): View {
        binding.searchInput.isFocusable = true
        binding.searchInput.isIconified = false
        binding.searchInput.maxWidth = Int.MAX_VALUE
        binding.searchInput.inputType = EditorInfo.TYPE_CLASS_TEXT
        binding.searchInput.isSubmitButtonEnabled = false
        binding.searchInput.queryHint = searchHintString
        binding.searchInput.setSearchHintTextColor(getThemedColor(context, R.attr.color_group_63))
        binding.searchInput.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(s: String): Boolean {
                return false
            }

            override fun onQueryTextChange(s: String): Boolean {
                binding.searchInput.setCloseButtonVisibility(s)
                callback.onQueryTextChange(s)
                return true
            }
        })
        binding.searchInput.setOnQueryTextFocusChangeListener { _: View?, isFocus: Boolean ->
            if (!isFocus) {
                callback.onQueryTextFocusChange()
            }
        }

        // remove focus line from search plate
        val searchEditPlate = binding.searchInput.findViewById<View>(androidx.appcompat.R.id.search_plate)
        searchEditPlate.setBackgroundColor(Color.TRANSPARENT)
        showSoftKeyboard(binding.searchInput)
        return binding.root
    }

    override fun overridesItemVisibility(): Boolean {
        return true
    }

    fun selectAllQueryTexts() {
        binding.searchInput.selectAllQueryTexts()
    }
}
